recipes/newrelic/infrastructure/ohi/memcached/debian.yml

```diff
@@ -81,49 +66,117 @@ install:
     setup:
       label: "Installing Memcached integration..."
       cmds:
+        - sudo apt-get install libmemcached-tools -y > /dev/null
         - |
           sudo mkdir -p "/etc/newrelic-infra/integrations.d"
         - |
           # Get latest definitions and skip any failure because of deprecation
           sudo apt-get -o Acquire::Check-Valid-Until=false update -yq
         - |
-          sudo apt-get install nri-memcached -y
+          sudo apt-get install nri-memcached -y > /dev/null
         - |
           if [ -f /etc/newrelic-infra/integrations.d/memcached-config.yml ]; then
             sudo rm /etc/newrelic-infra/integrations.d/memcached-config.yml;
           fi
         - |
-          if [ -z {{.NR_CLI_DB_USERNAME}} ] && [ -z {{.NR_CLI_DB_PASSWORD}} ]; then
-            sudo tee /etc/newrelic-infra/integrations.d/memcached-config.yml > /dev/null <<"EOT"
+          
+          # try get values from env
+          NR_CLI_DB_USERNAME={{.NR_CLI_DB_USERNAME}}
+          NR_CLI_DB_PASSWORD={{.NR_CLI_DB_PASSWORD}}
+          NR_CLI_HOST={{.NR_CLI_HOST}}
+          NR_CLI_PORT={{.NR_CLI_PORT}}
+
+          # Set default values for variables
+          NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+          NR_CLI_PORT=${NR_CLI_PORT:-11211}
+          MAX_RETRIES=3
+          TRIES=0 
+          CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+          while [[ $TRIES -lt $MAX_RETRIES  && ( $CAN_CONNECT == "6" || $CAN_CONNECT == "7" ) ]]; do
+            if [ $CAN_CONNECT == "6" ]; then 
+              printf "\nMemcached Hostname (default: localhost): "
+              read -r NR_CLI_HOST
+              NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+            fi
+
+            printf "\nMemcached Port (default: 11211): "
+            read -r NR_CLI_DB_PORT
+            NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-11211}
+            ((TRIES++))
+
+            CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+            if [ $CAN_CONNECT == "6" ]; then
+              printf "\n[Error]: Provided host ($NR_CLI_DB_HOSTNAME) cannot be reached.\nSee https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/postgresql-monitoring-integration#comp-req for more info.\n" >&2
+              if [ ! $TRIES -lt $MAX_RETRIES ]; then exit 6; fi
+              echo "Please try again"
+            elif [ $CAN_CONNECT == "7" ]; then
+              printf "\n[Error]: Connections to port ($NR_CLI_DB_PORT) are being refused.\nSee https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/postgresql-monitoring-integration#comp-req for more info.\n" >&2
```

Comment:
@NRhzhao maybe we can craft a better message when we see we can't connect with the port, otherwise this sounds a bit alarming as a user, particularly since this is what users are going to see before been able to specify a port when the default port is not used (They havent had a chance to input that information yet).
Maybe something like "We can't connect to the rabbitmq on localhost using the port 11211, please enter the port number of your rabbitmq instance".

Reply:
Updated